lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php

```diff
@@ -82,16 +82,16 @@ public function execute(InputInterface $input, OutputInterface $output)
             $path = is_bool($path) ? getcwd() : $path;
             $version->writeSqlFile($path, $direction);
         } else {
-            $noInteraction = $input->getOption('no-interaction') ? true : false;
-            if ($noInteraction === true) {
-                $version->execute($direction, $input->getOption('dry-run') ? true : false);
+            if ($input->getOption('no-interaction')) {
```

Comment:
I'm a bit nervous as this changes functionality slightly - I've seen people use constructs like `echo y | app/console doctrine:migrations:migrate` because they didn't know about the `--no-interaction` option - that kind of thing might break. I'd rather put that in a separate PR from this, which is just intended to tidy some things a bit.